components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -660,84 +681,110 @@ internal sealed class SubscriptionWrapper : ISubscription {
+                                _subscription.Id,
+                                _subscription.Connection.CreateConnectionId(),
+                                configuration?.MaxNotificationsPerPublish ?? 0);
+                            subscription.MaxNotificationsPerPublish =
+                                _subscription.Configuration.MaxNotificationsPerPublish.GetValueOrDefault(0);
+                            modifySubscription = true;
+                        }
+                        if (subscription.LifetimeCount != _subscription.Configuration.LifetimeCount
+                                .GetValueOrDefault(session.DefaultSubscription.LifetimeCount)) {
+                            _logger.Debug(
+                                "Subscription '{subscription}'/'{sessionId}' change LifetimeCount to {new}",
+                                _subscription.Id,
+                                _subscription.Connection.CreateConnectionId(),
+                                configuration?.LifetimeCount ?? 0);
+                            subscription.LifetimeCount = _subscription.Configuration.LifetimeCount
+                                .GetValueOrDefault(session.DefaultSubscription.LifetimeCount);
+                            modifySubscription = true;
+                        }
+                        if (subscription.Priority != _subscription.Configuration.Priority
+                                .GetValueOrDefault(session.DefaultSubscription.Priority)) {
+                            _logger.Debug("Subscription '{subscription}'/'{sessionId}' change Priority to {new}",
+                                _subscription.Id,
+                                _subscription.Connection.CreateConnectionId(),
+                                configuration?.Priority ?? 0);
+                            subscription.Priority = _subscription.Configuration.Priority
+                                .GetValueOrDefault(session.DefaultSubscription.Priority);
+                            modifySubscription = true;
+                        }
+                        if (modifySubscription) {
+                            subscription.Modify();
+                            logRevisedValues = true;
+                        }
                     }
                     if (subscription.CurrentPublishingEnabled != activate) {
                         // do not deactivate an already activated subscription
                         subscription.SetPublishingMode(activate);
+                        logRevisedValues = true;
                     }
                 }
+
+                if (logRevisedValues && activate) {
+                    var diagInfo = new StringBuilder();
+                    diagInfo.Append("Subscription '{subscription}'/'{sessionId}' state actual(revised)/desired: ");
+                    diagInfo.Append("PublishingEnabled {currentPublishingEnabled}/{publishingEnabled} ");
```

Comment:
Should we add commas after reporting expected and actual values for each field ?